docs/design/2023-06-30-configurable-kv-timeout.md

```diff
@@ -0,0 +1,238 @@
+# Proposal: Configurable KV Timeout
+
+* Authors: [cfzjywxk](https://github.com/cfzjywxk)
+* Tracking issue: TODO
+
+## Motivation
+
+The associated TiKV requests are typically processed quickly, taking only a few milliseconds. However, if 
+there is disk IO or network latency jitter on a specified TiKV node, the duration of these requests may spike 
+to several seconds or more. This usually happens when there is an EBS issue like the EBS is repairing its 
+replica.
```

Comment:
Shall we make it clear that it should only affect requests that have more than 1 candidate target?

Reply:
Added at the end of this section.